nix/0001-Support-per-installation-tokens-in-GithubStatusPush-.patch

```diff
@@ -0,0 +1,66 @@
+From 65dd3e63dc0dc34a531eee703baff715ba04d0a0 Mon Sep 17 00:00:00 2001
+From: magic_rb <richard@brezak.sk>
+Date: Wed, 10 Jul 2024 13:00:48 +0200
+Subject: [PATCH 1/3] Support per installation tokens in `GithubStatusPush`
+ reporter
+
+Signed-off-by: magic_rb <richard@brezak.sk>
+---
+ master/buildbot/reporters/github.py | 20 +++++++++++++++++---
+ 1 file changed, 17 insertions(+), 3 deletions(-)
+
+diff --git a/master/buildbot/reporters/github.py b/master/buildbot/reporters/github.py
+index 3873c2676..e61e6495a 100644
+--- a/master/buildbot/reporters/github.py
++++ b/master/buildbot/reporters/github.py
+@@ -70,7 +70,14 @@ class GitHubStatusPush(ReporterBase):
+         generators=None,
+         **kwargs,
+     ):
+-        token = yield self.renderSecrets(token)
++        headers = {}
++        if not callable(token):
++            token = yield self.renderSecrets(token)
```

Comment:
I think the renderSecrets interface would be an alternative approach to this actually. One can also define an object that would the same thing we are currently doing.

Reply:
Mhm. But than it's not per installation...